applications/teletype/src/teletype_fax_util.erl

```diff
@@ -213,47 +212,30 @@ maybe_fetch_attachments(_, _, _, 'true') ->
 maybe_fetch_attachments(DataJObj, FaxJObj, Macros, 'false') ->
     FaxId = kz_doc:id(FaxJObj),
     Db = kz_doc:account_db(FaxJObj),
-    [AttachmentName] = kz_doc:attachment_names(FaxJObj),
-
-    lager:debug("accessing fax attachment ~s at ~s / ~s", [AttachmentName, Db, FaxId]),
+    lager:debug("accessing fax attachment ~s at ~s", [Db, FaxId]),
     teletype_util:send_update(DataJObj, <<"pending">>),
-
-    case kz_datamgr:fetch_attachment(Db, {kzd_fax:type(), FaxId}, AttachmentName) of
-        {'ok', Bin} ->
-            ContentType = kz_doc:attachment_content_type(FaxJObj, AttachmentName),
-            maybe_convert_attachment(DataJObj, Macros, ContentType, Bin);
+    Format = kapps_config:get_ne_binary(?CONVERT_CONFIG_CAT, [<<"fax">>, <<"attachment_format">>], <<"pdf">>),
+    Filename = get_file_name(Macros, <<".", Format/binary>>),
+    case fetch_attachment(Format, Db, FaxJObj) of
+        {'ok', Content, ContentType, _Doc} ->
+            [{ContentType, Filename, Content}];
         {'error', _E} ->
-            lager:debug("failed to fetch attachment ~s: ~p", [AttachmentName, _E]),
-            []
-    end.
-
--spec maybe_convert_attachment(kz_json:object(), kz_term:proplist(), kz_term:api_binary(), binary()) -> attachments().
-maybe_convert_attachment(DataJObj, Macros, ContentType, Bin) ->
-    ToFormat = kapps_config:get_ne_binary(?FAX_CONFIG_CAT, <<"attachment_format">>, <<"pdf">>),
-    FromFormat = from_format_from_content_type(ContentType),
-
-    case convert(DataJObj, FromFormat, ToFormat, Bin) of
-        {'ok', Converted} ->
-            Filename = get_file_name(Macros, ToFormat),
-            lager:debug("adding attachment as ~s", [Filename]),
-            [{content_type_from_extension(Filename), Filename, Converted}];
-        {'error', Reason} ->
-            lager:debug("error converting attachment with reason : ~p", [Reason]),
+            lager:debug("failed to fetch attachment: ~p", [_E]),
             []
     end.
 
--spec from_format_from_content_type(kz_term:ne_binary()) -> kz_term:ne_binary().
-from_format_from_content_type(<<"application/pdf">>) -> <<"pdf">>;
-from_format_from_content_type(<<"image/tiff">>) -> <<"tif">>;
-from_format_from_content_type('undefined') -> <<"tif">>;
-from_format_from_content_type(ContentType) ->
-    [_Type, SubType] = binary:split(ContentType, <<"/">>),
-    SubType.
-
--spec content_type_from_extension(kz_term:ne_binary()) -> kz_term:ne_binary().
-content_type_from_extension(Ext) ->
-    {Type, SubType, _} = cow_mimetypes:all(Ext),
-    <<Type/binary, "/", SubType/binary>>.
+-spec fetch_attachment(kz_term:ne_binary(), kz_term:ne_binary(), kz_json:object()) ->
+                              {'ok', kz_term:ne_binary(), kz_term:ne_binary(), kz_json:object()} |
+                              {'error', any()}.
+fetch_attachment(<<"original">>, Db, Doc) ->
```

Comment:
hmm, i think you have this function in crossbar too. So it is better to move it into `kzd_fax`, pass the fax_id, db and desired format and it call the appropriate function.